app/src/main/java/com/arctouch/codechallenge/util/recycler_view/EndlessRecyclerOnScrollListener.java

```diff
@@ -0,0 +1,44 @@
+package com.arctouch.codechallenge.util.recycler_view;
+
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+public abstract class EndlessRecyclerOnScrollListener extends RecyclerView.OnScrollListener {
```

Comment:
Why did you choose to use Java for this `class`?

Reply:
I took that code from another source here in github by the user "Tianxing Li" or "nex3z". So I didn't stop to convert the code for kotlin. And sorry for not putting this in the comments.